test/controllers/surveys_controller_test.rb

```diff
@@ -30,7 +35,9 @@ class SurveysControllerTest < ActionDispatch::IntegrationTest
   end
 
   test "show page for unpublished survey should redirect to root" do
-    assert false
+    sign_in @user
+    post surveys_url, params: { survey: { name: @survey.name, published: @survey.published, user_id: @survey.user_id } }
+    assert_redirected_to root_path
```

Comment:
This tests creating a survey, not the show page

Reply:
Good point @spencerldixon! I will fix this.